code/game/machinery/computer/arcade/orion.dm

```diff
@@ -0,0 +1,524 @@
+	var/killed_crew = 0
+
+
+/obj/machinery/computer/arcade/orion_trail/Initialize()
+	. = ..()
+	Radio = new /obj/item/radio(src)
+	Radio.listening = 0
+	for(var/path in events)
+		var/datum/orion_event/new_event = new path(src)
+		events[new_event] = new_event.weight
+		popleft(events) //remove the old path
+
+/obj/machinery/computer/arcade/orion_trail/Destroy()
+	QDEL_NULL(Radio)
+	for(var/datum/orion_event/dat_to_del as anything in events)
+		dat_to_del.game = null
+		qdel(dat_to_del)
+	return ..()
+
+/obj/machinery/computer/arcade/orion_trail/kobayashi
+	name = "Kobayashi Maru control computer"
+	desc = "A test for cadets."
+	icon = 'icons/obj/machines/particle_accelerator.dmi'
+	icon_state = "control_boxp"
+	events = list("Raiders" = 3, "Interstellar Flux" = 1, "Illness" = 3, "Breakdown" = 2, "Malfunction" = 2, "Collision" = 1, "Spaceport" = 2)
+	prize_override = list(/obj/item/paper/fluff/holodeck/trek_diploma = 1)
+	settlers = list("Kirk","Worf","Gene")
+
+/obj/machinery/computer/arcade/orion_trail/proc/newgame()
+	// Set names of settlers in crew
+	var/mob/living/carbon/player = usr
+	var/player_crew_name = player.first_name()
+	settlers = list()
+	for(var/i = 1; i <= 3; i++)
+		add_crewmember(update = FALSE)
+	add_crewmember("[player_crew_name]") //final crewmember is you, so DO update the settler moods now
+	// Re-set items to defaults
+	engine = 1
+	hull = 1
+	electronics = 1
+	food = 80
+	fuel = 60
+	alive = 4
+	turns = 1
+	event = null
+	gameStatus = ORION_STATUS_NORMAL
+	lings_aboard = 0
+	lings_suspected = FALSE
+	killed_crew = 0
+
+	//spaceport junk
+	spaceport_raided = FALSE
+
+/obj/machinery/computer/arcade/orion_trail/proc/report_player(mob/gamer)
+	if(gamers[gamer] == -2)
+		return // enough harassing them
+
+	if(gamers[gamer] == -1)
+		say("WARNING: Continued antisocial behavior detected: Dispensing self-help literature.")
+		new /obj/item/paper/pamphlet/violent_video_games(drop_location())
+		gamers[gamer]--
+		return
+
+	if(!(gamer in gamers))
+		gamers[gamer] = 0
+
+	gamers[gamer]++ // How many times the player has 'prestiged' (massacred their crew)
+
+	if(gamers[gamer] > 2 && prob(20 * gamers[gamer]))
+
+		Radio.set_frequency(FREQ_SECURITY)
+		Radio.talk_into(src, "SECURITY ALERT: Crewmember [gamer] recorded displaying antisocial tendencies in [get_area(src)]. Please watch for violent behavior.", FREQ_SECURITY)
+
+		Radio.set_frequency(FREQ_MEDICAL)
+		Radio.talk_into(src, "PSYCH ALERT: Crewmember [gamer] recorded displaying antisocial tendencies in [get_area(src)]. Please schedule psych evaluation.", FREQ_MEDICAL)
+
+		gamers[gamer] = -1
+
+		gamer.client.give_award(/datum/award/achievement/misc/gamer, gamer) // PSYCH REPORT NOTE: patient kept rambling about how they did it for an "achievement", recommend continued holding for observation
+		gamer.mind?.adjust_experience(/datum/skill/gaming, 50) // cheevos make u better
+
+		if(!isnull(GLOB.data_core.general))
+			for(var/datum/data/record/R in GLOB.data_core.general)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names